services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5396,21 +5396,28 @@ private Pair<ExecutionStatus, String> getDaVinciPushStatusAndDetails(
     Optional<String> erroredInstance = Optional.empty();
     String storeName = version.getStoreName();
     int completedPartitions = 0;
+    int totalInstanceCount = 0;
+    int liveInstanceCount = 0;
     Set<Integer> incompletePartition = new HashSet<>();
     for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
       Map<CharSequence, Integer> instances = pushStatusStoreReader.get()
           .getPartitionStatus(storeName, version.getNumber(), partitionId, incrementalPushVersion);
       boolean allInstancesCompleted = true;
+      totalInstanceCount += instances.size();
       for (Map.Entry<CharSequence, Integer> entry: instances.entrySet()) {
         ExecutionStatus status = ExecutionStatus.fromInt(entry.getValue());
+        boolean isInstanceAlive = pushStatusStoreReader.get().isInstanceAlive(storeName, entry.getKey().toString());
+
+        if (isInstanceAlive) {
+          liveInstanceCount++;
+        }
         if (status == middleStatus) {
-          if (allInstancesCompleted
-              && pushStatusStoreReader.get().isInstanceAlive(storeName, entry.getKey().toString())) {
+          if (allInstancesCompleted && isInstanceAlive) {
             allInstancesCompleted = false;
           }
```

Comment:
Interesting; so we have a very strict requirement already, that all the DVC node must be up and running. Does it mean during the incident, DVC node didn't crash during the push job, but after the push job? Can we revisit that incident?

Reply:
Talked with Sourav offline; we didn't have the log anymore.